public/apps/login/login-page.tsx

```diff
@@ -49,16 +50,34 @@ interface LoginButtonConfig {
   buttonstyle: string;
 }
 
-function redirect(serverBasePath: string) {
-  // navigate to nextUrl
+export function getNextPath(serverBasePath: string) {
   const urlParams = new URLSearchParams(window.location.search);
   let nextUrl = urlParams.get('nextUrl');
   if (!nextUrl || nextUrl.toLowerCase().includes('//')) {
     // Appending the next url with trailing slash. We do so because in case the serverBasePath is empty, we can simply
     // redirect to '/'.
     nextUrl = serverBasePath + '/';
   }
-  window.location.href = nextUrl + window.location.hash;
+  const savedTenant = getSavedTenant();
+  const url = new URL(
+    window.location.protocol + '//' + window.location.host + nextUrl + window.location.hash
+  );
+  if (
+    !!savedTenant &&
+    !(
+      url.searchParams.has('security_tenant') ||
+      url.searchParams.has('securitytenant') ||
+      url.searchParams.has('securityTenant_')
+    )
+  ) {
```

Comment:
nit: 
```ts
if (shouldTenantInfoBeAddedToUrl(savedTenant, url) {
```

and a method:
```ts
function shouldTenantInfoBeAddedToUrl(savedTenant: string, url: object) {
    const tenantIsSaved = !!savedTenant;
    const hasSecurityTenantParam = url.searchParams.has('security_tenant') || url.searchParams.has('securitytenant') || url.searchParams.has('securityTenant_');

    return tenantIsSaved && !hasSecurityTenantParam;
}
```

Reply:
When I switch the condition here, VS Code underlines `savedTenant` in the call to add it as a param (` url.searchParams.append('security_tenant', savedTenant);`) since its required to be non-null when added as a param. Even though there is a null check in `shouldTenantInfoBeAddedToUrl`, the IDE doesn't pick up on it.

Reply:
hm, that's interesting. I guess it is fine as is then.